src/ucp/core/ucp_ep.h

```diff
@@ -42,16 +47,17 @@ enum {
                                                         (on server side due to receiving partial
                                                         worker address from the client) */
     UCP_EP_FLAG_CONNECT_PRE_REQ_QUEUED = UCS_BIT(9), /* Pre-Connection request was queued */
+    UCP_EP_FLAG_CLOSED                 = UCS_BIT(10),/* EP was closed */
```

Comment:
extra flag, you can just disable `UCP_EP_FLAG_USED` flag on closure

Reply:
the problem is that there was requirement that message which arrive *before* an internal endpoint is "retrieved" will not be dropped

Reply:
Then EP should be on match queue, right?

Reply:
i disable UCP_EP_FLAG_USED on closure, it means stream recv has to drop packets if UCP_EP_FLAG_USED is not set. But it's also not set when ep is on match queue and before it was retrieved by ucp_ep_create, and we don't want this to happen

Reply:
forget it :) i'm thinking in scope of not implemented FIN protocol.